lib/components/Loader/ProgressRing.tsx

```diff
@@ -59,16 +60,30 @@ const ProgressRing: FC<ProgressRingProps> = ({
   const circumference = normalizedRadius * 2 * Math.PI;
   const strokeDashoffset = circumference - (percentage / 100) * circumference;
 
-  const getRingColor = () => {
-    return colorGradient ? "url(#ring_gradient)" : theme.colors.primary[500];
+  const getRingForegroundColor = () => {
+    if (colorGradient) {
+      return "url(#ring_gradient)";
+    }
+    return variant
+      ? theme.progressRing.ringForegroundColor[variant]
+      : theme.progressRing.ringForegroundColor.primary;
+  };
+
+  const getRingBackgroundColor = () => {
+    if (colorGradient) {
+      return theme.progressRing.ringBackgroundColor.primary;
+    }
+    return variant
+      ? theme.progressRing.ringBackgroundColor[variant]
+      : theme.progressRing.ringBackgroundColor.primary;
```

Comment:
useMemo

Reply:
Thanks for the review, I will add the tests and implement useMemo.